src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/virologischerbefund/BefundJedesEreignisPointEventConverter.java

```diff
@@ -0,0 +1,110 @@
+package org.ehrbase.fhirbridge.ehr.converter.specific.virologischerbefund;
+
+import ca.uhn.fhir.rest.server.exceptions.UnprocessableEntityException;
+import org.ehrbase.fhirbridge.ehr.converter.generic.ObservationToPointEventConverter;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.BefundJedesEreignisPointEvent;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.LabortestPanelCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProbeCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.AnatomischeLokalisationCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.NameDerKoerperstelleDefiningCode;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.LabortestBezeichnungDefiningCode;
+
+import org.hl7.fhir.exceptions.FHIRException;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Specimen;
+
+import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BefundJedesEreignisPointEventConverter extends ObservationToPointEventConverter<BefundJedesEreignisPointEvent> {
+
+    @Override
+    protected BefundJedesEreignisPointEvent convertInternal(Observation observation) throws FHIRException {
+
+        Specimen specimen = observation.getSpecimenTarget();
+
+        BefundJedesEreignisPointEvent befundevent = new BefundJedesEreignisPointEvent();
+
+        mapLabortestbezeichnung(observation, befundevent);
+        mapProbe(befundevent, specimen);
+
+        List<ProAnalytCluster> proAnalytClusterlist = new ArrayList<>();
+        LabortestPanelCluster labortestPanelCluster = new LabortestPanelCluster();
+
+        proAnalytClusterlist.add(new ProAnalytClusterConverter().convert(observation));
+        labortestPanelCluster.setProAnalyt(proAnalytClusterlist);
+        befundevent.setLabortestPanel(labortestPanelCluster);
+
+        return befundevent;
+
+    }
+
+    private void mapLabortestbezeichnung(Observation observation, BefundJedesEreignisPointEvent befundevent) {
+
+        if(observation.getCategory().get(0).getCoding().get(2).getCode().equals("122442008")) {
+            befundevent.setLabortestBezeichnungDefiningCode(LabortestBezeichnungDefiningCode.DETECTION_OF_VIRUS_PROCEDURE);
+        }else{
+            throw new UnprocessableEntityException("createLabortestBezeichnungDefiningCode failed.");
+        }
+    }
+
+    private void mapProbe(BefundJedesEreignisPointEvent befundevent, Specimen specimen) throws FHIRException {
```

Comment:
Method too long please split in smaller methods

Reply:
Simplifier Link: https://simplifier.net/medizininformatikinitiative-highmed-ic/virologischerbefund
Highmed Link: http://88.198.146.13/ckm/templates/1246.169.636

I used the canonical link (http://highmed.org/StructureDefinition/VirologischerBefund) in the Structure Definition of the simplifier site, but apparently it is still in draft and can still change. It hasn't updated to a new one on the simplifier site yet.

Reply:
As I just saw that the canonical links for the "Diagnostic Report" and "Specimen"  also changed I will post them here as well. 
Diagnostic Report: https://simplifier.net/coreprofilesstu3/diagnosticreport
Specimen: https://simplifier.net/coreprofilesstu3/specimen

Reply:
Okay, before i start to review, can you ensure again that the cardinality is always checked if the field in fhir is not a most have ? Since this is very often a bug people tend to do, would do me a great favour, since we are currently super busy. 
Thanks.
I will start the review as soon as i have time (as you see currently we are doing a great deal of stuff at the bridge specifically for gecco).

Reply:
I don't think I have an extra cardinality check for every optional fhir rescource and it would just throw a standard error when trying to access it. 
As a result, I will go through the code again and implement a "if not present throw exception" check on every used optional fhir rescource when it shows up.
I can also combine this with updating the outdated links and splitting up that method you already mentioned. Just need to ask in a meeting tomorrow which links I can reliably use. I probably should be able to commit and push the adjustments tomorrow.

Reply:
I would suggest, if the field ist not necessary in openEHR, you dont map it (use return Optional.empty()) 
Example: 
https://github.com/ehrbase/fhir-bridge/blob/c1aec64a9e1f677330d247720674ebeb44c346c3/src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/bodyheight/GroesseLaengeObservationConverter.java#L19

If the code/value is not valid you need to throw an:
ConversionException. 

This can be sometimes a pain in the arse, i know, but it has to be done, otherwise the bridge will crash with an normal exception, this is not what we want. Remember if the field is REQUIRED in fhir you dont need to do this check, since HAPI does it :)

Thanks